src/server/webapi/objects.cpp

```diff
@@ -267,37 +270,41 @@ bool HasDescendantsMatchingClassFilter(uint32_t objectId, uint32_t userId, const
 /**
  * Recursively build nested object tree structure
  */
-json_t *BuildNestedObjectTree(uint32_t objectId, uint32_t userId, const std::unordered_set<int> &classFilter, std::unordered_set<uint32_t> &visited, int maxDepth, int currentDepth)
+static json_t *BuildNestedObjectTree(uint32_t objectId, uint32_t userId, const std::unordered_set<int> &classFilter, std::unordered_set<uint32_t> &visited, int maxDepth, int currentDepth)
 {
    if (visited.count(objectId) > 0 || currentDepth >= maxDepth)
       return json_array();
 
    visited.insert(objectId);
 
-   unique_ptr<SharedObjectArray<NetObj>> children = g_idxObjectById.getObjects(
-      [userId, objectId, &classFilter, maxDepth, currentDepth] (NetObj *obj) -> bool
-      {
-         if (obj->isHidden() || obj->isSystem() || obj->isDeleted() || !obj->checkAccessRights(userId, OBJECT_ACCESS_READ))
-            return false;
+   shared_ptr<NetObj> object = FindObjectById(objectId);
```

Comment:
caller of this function already has pointer to object. It can be passed as argument instead of passing ID and finding object.